### PR TITLE
add quaternions/20160606

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **quaternions/20160606**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/quaternions/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    def package_id(self):
+        self.info.header_only()
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        url = self.conan_data["sources"][self.version]["url"]
+        extracted_dir = self.name + "-" + os.path.splitext(os.path.basename(url))[0]
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "include", "quaternion.h"),
+                              "#include <boost/mpl/bool.hpp>", "")
```
Comment (SSE4) [inline recipes/quaternions/all/conanfile.py:32]: does it have a dependency on boost?

Comment (SpaceIm) [reply recipes/quaternions/all/conanfile.py:32]: I don't see anything in headers using boost. I think it's a remanent include. It depends on boost for tests.

Comment (uilianries) [reply recipes/quaternions/all/conanfile.py:32]: interesting, do you know if the upstream knows about? it can be a "bug".

Comment (SpaceIm) [reply recipes/quaternions/all/conanfile.py:32]: It's most certainly a bug: https://github.com/ferd36/quaternions/pull/3

But looking at the last commit date, the library is not maintained anymore.

Comment (uilianries) [reply recipes/quaternions/all/conanfile.py:32]: > But looking at the last commit date, the library is not maintained anymore.

Sadly, sometimes it happens ....

**Pull request merged**